indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -440,7 +442,7 @@ private void manageInternalCritical(
             notifyStatus(task, taskStatus, taskStatus.getErrorMsg());
             continue;
           }
-          if (taskIsReady) {
+          if (taskIsReady && !isControllerTaskLimitReached(task.getType(), true)) {
```

Comment:
Should It just be limited to MSQ controllers or other job types as well. Maybe Take a json as a input ? where key is the taskType and the value is the limit/float. 

Also what is the user behavior  if the task is pending for launch due to limit. 
* Can we communicate why the task is not launching to the user.
* If the cluster is totally starved, does the controller get timed out eventually and removed from queue?